R/LSTconfidenceIntervals.R

```diff
@@ -0,0 +1,407 @@
+  } else {
+    confidenceContainer <- createJaspContainer()
+    # we set the dependencies on the container, this means that all items inside the container automatically have these dependencies
+    confidenceContainer$dependOn(c("mu", "sigma", "n", "confidenceIntervalInterval", "nReps", 
+                                   "treePlot"))
+    jaspResults[["confidenceContainer"]] <- confidenceContainer
+  }
+  return(confidenceContainer)
+}
+
+.simulateDatasets <- function(jaspContainer, options) {
+  if (!is.null(jaspContainer[["simulatedDatasets"]]))
+    return()
+  
+  listWithData <- lapply(1:options$nReps, function(x) rnorm(options$n,
+                                                            options$mu,
+                                                            options$sigma))
+  
+  jaspContainer[["simulatedDatasets"]] <- createJaspState(object = listWithData)
+  
+  return()
+}
+
+.computeConfidenceIntervals <- function(jaspContainer, options) {
+  if (!is.null(jaspContainer[["computedConfidenceIntervals"]]))
+    return()
+  
+  listWithData <- jaspContainer[["simulatedDatasets"]][["object"]]
+  
+  meanCI <- t(sapply(listWithData, function(x) c(mean(x), 
+                                                 t.test(x, conf.level = options$confidenceIntervalInterval)$conf.int[1:2])))
+  meanCI <- as.data.frame(meanCI)
+  colnames(meanCI) <- c("mean", "lower", "upper")
+  meanCI[["Repetition"]] <- 1:options$nReps
+  meanCI[["successfulCI"]] <- options$mu > meanCI[["lower"]] & options$mu < meanCI[["upper"]]
+  meanCI[["colorCI"]] <- ifelse(meanCI[["successfulCI"]], "A", "B")
+  meanCI[["sd"]] <- sapply(listWithData, sd)
+  meanCI[["n"]] <- sapply(listWithData, length)
+  
+  
+  jaspContainer[["computedConfidenceIntervals"]] <- createJaspState(object = meanCI)
+  
+  return()
+}
+
+.makeSummaryTable <- function(jaspContainer, options) {
+  if (!is.null(jaspContainer[["resultTable"]]))
+    return()
+  
+  resultTable <- createJaspTable(title = "Results Table")
+  jaspContainer[["resultTable"]] <- resultTable
+  
+  resultTable$addColumnInfo(name = "Repetitions", type = "integer")
+  resultTable$addColumnInfo(name = "mu", type = "number", title=gettext("μ"))
+  resultTable$addColumnInfo(name = "sigma", title=gettext("σ"), type = "number")
+  resultTable$addColumnInfo(name = "n", type = "integer")
+  resultTable$addColumnInfo(name = "coverage", type = "integer",title=gettext("Coverage"))
+  resultTable$addColumnInfo(name = "coveragePercentage", type = "number", format = "pc", title=gettext("Coverage %"))
+  
+  meanCI <- jaspContainer[["computedConfidenceIntervals"]][["object"]]
+  tableDat <- data.frame(mu = options$mu,
+                         sigma = options$sigma,
+                         Repetitions = options$nReps,
+                         n = options$n,
+                         coverage = sum(meanCI[["successfulCI"]]),
+                         coveragePercentage = sum(meanCI[["successfulCI"]]) / options$nReps)
```

Comment:
shouldn't this be
```suggestion
                         coveragePercentage = mean(meanCI[["successfulCI"]])
```
?